tools/install-tcmalloc.sh

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env bash
+#
+# Script to install libtcmalloc.so in the current WiredTiger git
+# repository.
+#
+# This is serves the same purpose, but is NOT THE SAME as:
+#
+# test/evergreen/tcmalloc_install_or_build.sh
+#
+# and the tcmalloc source version referenced here should be updated in
+# sync with that script.
+
+set -euf -o pipefail
+
+die() { echo "$@"; exit 1; }
+
+HERE=$(git rev-parse --show-toplevel) || die "FATAL Run in root of git workspace"
+[[ -f ${HERE}/CMakeLists.txt ]] || die "Does not look like a WT workspace - no CMakeLists.txt"
+[[ -d ${HERE}/TCMALLOC_LIB ]] && die "TCMALLOC_LIB already present"
+
+# Before building look for a prebuilt copy cached on this system.
+SRC_ID=mongo-SERVER-85737
+CACHED_DIR=${HOME}/.cache/wiredtiger/tcmalloc/${SRC_ID}
+CACHED_BIN=${HOME}/.cache/wiredtiger/tcmalloc/${SRC_ID}/TCMALLOC_LIB.tgz
+if [[ -r ${CACHED_BIN} ]]; then
+    echo "Installing cached copy from $CACHED_BIN"
+    tar zxf $CACHED_BIN
+    exit
+fi
+
+echo "No cached copy on this system. Attempting to build."
+
+# Requires bazel.
+which bazel || echo "FATAL Build tool bazel not found in path"
+
+# Use mongo toolchain
+[[ -d /opt/mongodbtoolchain/v4/bin ]] || die "FATAL Requires mongodb toolchain."
```

Comment:
Is mongodb toolchain a must or a nice-to-have for building tcmalloc? I'm wondering if it's possible to use this script on a local laptop where toolchain is not readily available (or available but in a directory other than the standard one).

Reply:
This script is intended to be for use by mongodb employed developers. If community developers want to work with tcmalloc they can load it from their respective environments.

Reply:
Thanks for clarifying it. I was actually thinking about internal users who use their laptops as the main development environment. Unfortunately, mongodb toolchain does not support macOS AFAIK. So they need to adjust this script before using it (which is fine).